lib/validation-strategies/openapi-v2.js

```diff
@@ -0,0 +1,741 @@
+            should(path.indexOf('{'+param.name+'}')).be.greaterThan(-1,'path parameters must appear in the path');
+        }
+    }
+    if (typeof param.required !== 'undefined') should(param.required).have.type('boolean');
+
+    if (param.in === 'body') {
+        param.should.have.property('schema');
+        checkSchema(param.schema, emptySchema, 'schema', openapi, options);
+    }
+
+    if (param.in !== 'body') {
+        param.should.have.property('type');
+        if (param.type === 'array') {
+            param.should.have.property('items');
+            param.items.should.be.type('object');
+        }
+    }
+
+    if (param.description) {
+        param.description.should.have.type('string');
+    }
+
+    lintObject('parameter',param,options);
+    options.context.pop();
+    return param;
+}
+
+function checkSecurity(security,openapi,options) {
+    contextAppend(options, 'security');
+    security.should.be.an.Array();
+    for (let sr of security) {
+        sr.should.be.an.Object();
+        sr.should.not.be.an.Array();
+        for (let i in sr) {
+            sr[i].should.be.an.Array();
+            let sec = jptr.jptr(openapi,'#/securityDefinitions/'+i);
+            sec.should.not.be.exactly(false,'Could not dereference securityDefinition '+i);
+            if (sec.type !== 'oauth2') {
+                sr[i].should.be.empty();
+            }
+            else if (sr[i].length) {
+                if (!options.allScopes[i]) {
+                    options.allScopes[i] = {};
+
+                    if (sec.flow == "password") Object.assign(options.allScopes[i],sec.scopes);
+                    if (sec.flow == "implicit") Object.assign(options.allScopes[i],sec.scopes);
+                    if (sec.flow == "authorizationCode") Object.assign(options.allScopes[i],sec.scopes);
+                    if (sec.flow == "clientCredentials") Object.assign(options.allScopes[i],sec.scopes);
+                }
+                for (let scope of sr[i]) {
+                    options.allScopes[i].should.have.property(scope);
+                }
+            }
+        }
+    }
+    lintObject('security',security,options);
+    options.context.pop();
+}
+
+function validateSync(openapi, options, callback) {
+    setupOptions(options,openapi);
+
+    should(openapi).be.an.Object();
+    openapi.should.not.have.key('openapi');
+    openapi.should.have.key('swagger');
+    openapi.swagger.should.have.type('string');
+    should.ok(openapi.swagger.startsWith('2.'), 'Must be an OpenAPI 2.x document');
+    openapi.should.have.key('paths');
+    should(openapi.paths).be.an.Object();
+
+    for (let k in openapi) {
+        if (!k.startsWith('x-')) {
+            should(['swagger', 'info', 'host', 'basePath', 'schemes', 'consumes','produces','paths', 'definitions','parameters','responses', 'securityDefinitions', 'security', 'tags', 'externalDocs'].indexOf(k)).be.greaterThan(-1,'OpenAPI object cannot have additionalProperty: '+k);
```

Comment:
Are the `produces` and `consumes` arrays themselves validated?

Reply:
I've added validation of the produces and consumes in my last commit.